dev/html/public/optimized-appear/defer-handoff-external-values.html

```diff
@@ -0,0 +1,119 @@
+<html>
+    <head>
+        <style>
+            body {
+                padding: 100px;
+                margin: 0;
+            }
+
+            #box {
+                width: 100px;
+                height: 100px;
+                background-color: #0077ff;
+            }
+
+            [data-layout-correct="false"] {
+                background: #dd1144 !important;
+                opacity: 1 !important;
+            }
+        </style>
+    </head>
+    <body>
+        <div id="root"></div>
+        <script type="module" src="/src/imports/optimized-appear.js"></script>
+        <script type="module" src="/src/imports/script-assert.js"></script>
+
+        <script type="module">
+            const {
+                motion,
+                animateStyle,
+                animate,
+                startOptimizedAppearAnimation,
+                optimizedAppearDataAttribute,
+                motionValue,
+                frame,
+            } = window.Motion
+            const { matchViewportBox } = window.Assert
+            const root = document.getElementById("root")
+
+            const duration = 0.5
+            const x = motionValue(0)
+            const animateX = motionValue(0)
+
+            let isFirstFrame = true
+
+            function Component() {
+                React.useEffect(() => {
+                    setTimeout(() => {
+                        x.set(200)
+                    }, 200)
+                }, [])
+
+                return React.createElement(motion.div, {
+                    id: "box",
+                    initial: { x: 0, opacity: 0 },
+                    animate: { x: 100, opacity: 1 },
+                    transition: {
+                        duration,
+                        ease: "linear",
+                        layout: { ease: () => 0, duration: 10 },
+                    },
+                    style: {
+                        x,
+                        position: "relative",
+                        background: "blue",
+                    },
+                    values: { x: animateX },
+                    onAnimationComplete: () => {
+                        const box = document.getElementById("box")
+                        const { left } = box.getBoundingClientRect()
+
+                        if (Math.round(left) !== 300) {
+                            showError(
+                                box,
+                                `optimised animation not cancelled by external value mismatch with rendered style`
+                            )
+                        }
```

Comment:
The opacity animation should not be canceled, right? Do we also want to check that as well?

Reply:
Yeah I'll add a check - it should be cancelled by mismatches as well